src/Server.php

```diff
@@ -11,16 +11,19 @@ class Server extends EventEmitter implements ServerInterface
 {
     private $io;
 
-    public function __construct(SocketServerInterface $io)
+    private $maxHeaderSize;
+
+    public function __construct(SocketServerInterface $io, $maxHeaderSize = 4096 /* 4 KiB */)
```

Comment:
The `maxHeaderSize` is not a property of the `Server` (this violates a SOLID design/DDD).

Ideally, we should inject a `RequestHeaderParser` via DI to the server (and make it default to the current behavior).

Unfortunately, the `RequestHeaderParser` is stateful, so we can't reuse it for multiple (concurrent) client connections. 

Perhaps we should consider passing a single instance to the ctor and using a clone thereof for each connection?

Arguably, this isn't necessarily part of this changeset. However, given that your changes introduce the necessity to pass this in the first place, this might make sense to discuss here.

Any thoughts?